projects/rero/ng-core/src/lib/record/editor/utils.ts

```diff
@@ -14,7 +14,102 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+import { FormlyFieldConfig } from '@ngx-formly/core';
 import { extractIdOnRef } from '../../utils/utils';
+import { ILogger } from '../../ILogger';
+
+/**
+ * Convert form to widget definition in JSONSchema
+ * @param schema - object, the JSONSchema
+ * @param logger - Logger
+ * @returns object, a converted JSONSchema
+ */
+export function formToWidget(schema: any, logger: ILogger): any {
+  if (schema.properties) {
+    for (const property of Object.keys(schema.properties)) {
+      formToWidget(schema.properties[property], logger);
+    }
+  }
+
+  /* items */
+  if (schema.items) {
+    formToWidget(schema.items, logger);
+  }
+
+  /* oneOf, anyOf, allOf */
+  ['oneOf', 'anyOf', 'allOf'].map((type: string) => {
+    if (schema[type]) {
+      schema[type].map((element: any) => formToWidget(element, logger));
+    }
+  });
+
+  /* form (custom) */
+  if (schema.form) {
```

Comment:
As you are a lot of code into this `if` statement, it seems more readable to invert the statement :
```javascript
if (!schema.form) {
    return schema
}
...
```

it will save a lot of empty spaces before each line included into the `if` statement